### PR TITLE
Update api.py

### PR DESCRIPTION
**Pull request by mayabrandi**
This PR fixes an error when a lims id is missing

## Description

Sometimes the `lims_id` does not exist in lims. Then the sample will be created but sample.udf will give an error:

```
>>> s=Sample(lims, id='hoho')
>>> s.udf
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/mayabrandi/opt/genologics/genologics/descriptors.py", line 350, in __get__
    instance.get()
  File "/Users/mayabrandi/opt/genologics/genologics/entities.py", line 289, in get
    self.root = self.lims.get(self.uri)
  File "/Users/mayabrandi/opt/genologics/genologics/lims.py", line 90, in get
    return self.parse_response(r)
  File "/Users/mayabrandi/opt/genologics/genologics/lims.py", line 209, in parse_response
    self.validate_response(response, accept_status_codes)
  File "/Users/mayabrandi/opt/genologics/genologics/lims.py", line 202, in validate_response
    raise requests.exceptions.HTTPError(message)
requests.exceptions.HTTPError: 404: Sample not found: hoho
```

**How to prepare for test**:
- [x] ssh to hasta (depending on type of change)
- [x] install on stage:
`bash servers/resources/hasta.scilifelab.se/update-cg-stage.sh fix_date_functions`

**How to test**:
Run following commands
- [x] `us`
- [x] `cg transfer lims --status prepared --sample-id latelychoiceemu`

**Expected test outcome**:
- [x] no error

**Review:**
- [x] code approved by @moonso 
- [x] tests executed by @mayabrandi 
- [x] "Merge and deploy" approved by @moonso 
Thanks for filling in who performed the code review and the test!

This [version](https://semver.org/) is a:

- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: cg/apps/lims/api.py [hunk 1]
```diff
@@ -93,25 +93,37 @@ def get_received_date(self, lims_id: str) -> str:
         """Get the date when a sample was received."""
 
         sample = Sample(self, id=lims_id)
-        return sample.udf.get("Received at") if sample else None
+        try:
+            return sample.udf.get("Received at")
+        except:
```
Comment (moonso) [inline cg/apps/lims/api.py:98]: Could you catch the expected error by importing in the top of the file:

```python
from requests.exceptions import HTTPError
```
Then 

```suggestion
        except HTTPError:
```
and same at the ones below

**Comment by moonso** [comment]
Hi @mayabrandi , I added unit tests now. It was a bit of a pain but at least I learned something new:) can you perform the manual tests on hasta an fill in above when you are ready and I will merge.

**Comment by mayabrandi** [comment]
You are an angel. Ill try to do that before the end of the day.

**Comment by mayabrandi** [comment]
Testing acording to above:

Running `cg transfer lims --status prepared --sample-id latelychoiceemu` on production gives error:
Skärmavbild 2020-05-29 kl  11 59 06

After installing this branch on stage, the same command on stage gives no error:

Skärmavbild 2020-05-29 kl  11 58 49

**Comment by moonso** [comment]
@barrystokman do you have time to help with merge and deploy?

**Comment by henrikstranneheim** [comment]
Well done! 👏

**Comment by moonso** [comment]
Sorry @barrystokman , lets wait a bit

**Comment by moonso** [comment]
Merged and deployed on hasta:

Skärmavbild 2020-05-29 kl  14 43 09

**Pull request merged**